src/ServiceControl.Monitoring/Metrics/Raw/OccurrenceSerializerDefinition.cs

```diff
@@ -0,0 +1,42 @@
+﻿namespace ServiceControl.Monitoring.Metrics.Raw
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using NServiceBus.MessageInterfaces;
+    using NServiceBus.Serialization;
+    using NServiceBus.Settings;
+
+    public class OccurrenceSerializerDefinition : SerializationDefinition
+    {
+        public override Func<IMessageMapper, IMessageSerializer> Configure(ReadOnlySettings settings)
+        {
+            return mapper => new OccurrenceSerializer();
+        }
+    }
+
+    public class OccurrenceSerializer : RawMessageSerializer<Occurrences>, IMessageSerializer
+    {
+        public object[] Deserialize(Stream stream, IList<Type> messageTypes = null)
+        {
+            long baseTicks = 0;
```

Comment:
I'm not sure about this being captured in an enclosure. Wondering if we should create a `state` struct and pass that around. It can include the creation of the message pool and set up the base ticks on the starter and then get passed in to each subsequent call.